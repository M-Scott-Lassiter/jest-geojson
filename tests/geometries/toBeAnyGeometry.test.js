// This matcher works on all the same geometry cases.
// Accordingly, those specific tests check that this function also work.
// This test suite also checks it fails with types Feature or FeatureCollection.
// Finally, it tests the unique snapshots.

const feature = {
    type: 'Feature',
    geometry: {
        type: 'Point',
        coordinates: [102.0, 0.5]
    }
}
const featureCollection = {
    type: 'FeatureCollection',
    features: [
        {
            type: 'Feature',
            geometry: {
                type: 'Point',
                coordinates: [102.0, 0.5]
            },
            properties: {
                prop0: 'value0'
            }
        },
        {
            type: 'Feature',
            geometry: {
                type: 'LineString',
                coordinates: [
                    [102.0, 0.0],
                    [103.0, 1.0],
                    [104.0, 0.0],
                    [105.0, 1.0]
                ]
            },
            properties: {
                prop0: 'value0',
                prop1: 0.0
            }
        },
        {
            type: 'Feature',
            geometry: {
                type: 'Polygon',
                coordinates: [
                    [
                        [100.0, 0.0],
                        [101.0, 0.0],
                        [101.0, 1.0],
                        [100.0, 1.0],
                        [100.0, 0.0]
                    ]
                ]
            },
            properties: {
                prop0: 'value0',
                prop1: {
                    this: 'that'
                }
            }
        }
    ]
}

describe('Valid Use Cases', () => {
    test('Expect to pass with GeometryCollection', () => {
        const geometryCollection = {
            type: 'GeometryCollection',
            geometries: [
                {
                    type: 'Point',
                    coordinates: [100.0, 0.0]
                },
                {
                    type: 'LineString',
                    coordinates: [
                        [101.0, 0.0],
                        [102.0, 1.0]
                    ]
                }
            ]
        }
        expect(geometryCollection).toBeAnyGeometry()
    })
})

describe('Invalid Use Cases', () => {
    test('Expect to fail with Feature', () => {
        expect(feature).not.toBeAnyGeometry()
    })

    test('Expect to fail with FeatureCollection', () => {
        expect(featureCollection).not.toBeAnyGeometry()
    })
})

describe('Error Snapshot Testing. Throws error:', () => {
    test('Valid use case passes', () => {
        expect(() =>
            expect({ type: 'Point', coordinates: [0, 0] }).not.toBeAnyGeometry()
        ).toThrowErrorMatchingSnapshot()
    })

    test('Invalid input to matcher', () => {
        expect(() => expect(false).toBeAnyGeometry()).toThrowErrorMatchingSnapshot()
    })

    test('Rejects Feature objects', () => {
        expect(() => expect(feature).toBeAnyGeometry()).toThrowErrorMatchingSnapshot()
        expect(() => expect(featureCollection).toBeAnyGeometry()).toThrowErrorMatchingSnapshot()
    })
})
