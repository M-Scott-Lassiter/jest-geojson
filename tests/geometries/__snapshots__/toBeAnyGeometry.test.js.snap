// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Snapshot Testing. Throws error: Invalid input to matcher 1`] = `
"[2mexpect([22m[31mGeometryObject[39m[2m).toBeAnyGeometry()[22m

Argument must be a GeoJSON Geometry object.

Received:  [31mfalse[39m"
`;

exports[`Error Snapshot Testing. Throws error: Rejects Feature objects 1`] = `
"[2mexpect([22m[31mGeometryObject[39m[2m).toBeAnyGeometry()[22m

Must have a type property with value 'Point', 'MultiPoint', 'LineString', 'MultiLineString', 'Polygon', 'MultiPolygon', or 'GeometryCollection'.

Received:  [31m{\\"geometry\\": {\\"coordinates\\": [102, 0.5], \\"type\\": \\"Point\\"}, \\"type\\": \\"Feature\\"}[39m"
`;

exports[`Error Snapshot Testing. Throws error: Rejects Feature objects 2`] = `
"[2mexpect([22m[31mGeometryObject[39m[2m).toBeAnyGeometry()[22m

Must have a type property with value 'Point', 'MultiPoint', 'LineString', 'MultiLineString', 'Polygon', 'MultiPolygon', or 'GeometryCollection'.

Received:  [31m{\\"features\\": [{\\"geometry\\": {\\"coordinates\\": [102, 0.5], \\"type\\": \\"Point\\"}, \\"properties\\": {\\"prop0\\": \\"value0\\"}, \\"type\\": \\"Feature\\"}, {\\"geometry\\": {\\"coordinates\\": [[102, 0], [103, 1], [104, 0], [105, 1]], \\"type\\": \\"LineString\\"}, \\"properties\\": {\\"prop0\\": \\"value0\\", \\"prop1\\": 0}, \\"type\\": \\"Feature\\"}, {\\"geometry\\": {\\"coordinates\\": [[[100, 0], [101, 0], [101, 1], [100, 1], [100, 0]]], \\"type\\": \\"Polygon\\"}, \\"properties\\": {\\"prop0\\": \\"value0\\", \\"prop1\\": {\\"this\\": \\"that\\"}}, \\"type\\": \\"Feature\\"}], \\"type\\": \\"FeatureCollection\\"}[39m"
`;

exports[`Error Snapshot Testing. Throws error: Valid use case passes 1`] = `
"[2mexpect([22m[31mGeometryObject[39m[2m).not.toBeAnyGeometry()[22m

Expected input to not be a valid GeoJSON geometry object.

Received:  [31m{\\"coordinates\\": [0, 0], \\"type\\": \\"Point\\"}[39m"
`;
